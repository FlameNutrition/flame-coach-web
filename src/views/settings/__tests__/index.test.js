import {
  render,
  screen
} from '../../../testing/test-utils';

import React from 'react';
import Settings from '../index';

describe('<Settings / >', () => {
  it('render component', async () => {
    const { container } = render(
      <Settings customerEmail="test@gmail.com" />
    );

    expect(container).toMatchSnapshot();
  });

  it('check if <Password /> component is called', async () => {
    render(
      <Settings customerEmail="test@gmail.com" />
    );

    expect(screen.queryAllByText('Previous Password')).not.toBeNull();
    expect(screen.queryAllByText('Password')).not.toBeNull();
    expect(screen.queryAllByText('Confirm password')).not.toBeNull();
  });
});
