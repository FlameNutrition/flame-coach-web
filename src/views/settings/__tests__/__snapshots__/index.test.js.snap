// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Settings / > render component 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg"
    >
      <div
        class="MuiBox-root MuiBox-root-2"
      >
        <form
          action="#"
        >
          <div
            class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              class="MuiCardHeader-root"
            >
              <div
                class="MuiCardHeader-content"
              >
                <span
                  class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                >
                  Password
                </span>
                <span
                  class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                >
                  Update password
                </span>
              </div>
            </div>
            <hr
              class="MuiDivider-root"
            />
            <div
              class="MuiCardContent-root"
            >
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                  data-shrink="false"
                >
                  Previous Password
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input"
                    name="oldPassword"
                    type="password"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-9"
                    >
                      <span>
                        Previous Password
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                  data-shrink="false"
                >
                  Password
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input"
                    name="newPassword"
                    type="password"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-9"
                    >
                      <span>
                        Password
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                  data-shrink="false"
                >
                  Confirm password
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input"
                    name="newPasswordConfirmation"
                    type="password"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-9"
                    >
                      <span>
                        Confirm password
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <hr
              class="MuiDivider-root"
            />
            <div
              class="MuiBox-root MuiBox-root-11 makeStyles-updateBtn-6"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                tabindex="0"
                type="submit"
              >
                <span
                  class="MuiButton-label"
                >
                  Update
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
